client/src/app/core/core-services/communication-manager.service.ts

```diff
@@ -34,23 +35,42 @@ class StreamHandler<T> {
 
     public closeCurrentStream(): void {
         const stream = this._currentActiveStream;
-        this._currentActiveStream?.close();
-        this._currentActiveStream = null;
+        this.destroyStream();
+        if (this._refreshTimer) {
+            clearInterval(this._refreshTimer);
+        }
         if (this._afterClosedFn) {
             this._afterClosedFn(stream);
         }
     }
 
     public openCurrentStream(): void {
-        if (!this._currentActiveStream) {
-            this.build();
+        this.reboot();
+        if (!this._refreshTimer) {
+            this._refreshTimer = setInterval(() => this.reboot(), 1000 * 60 * 30); // 30 min
```

Comment:
This is just a quick fix. We should discover alternatives, so a TCP connection won't break up